airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py

```diff
@@ -128,13 +146,24 @@ def test_two_sequential_reads(
         assert states_1, "Should produce at least one state"
         assert records_1, "Should produce at least one record"
 
-        latest_state = states_1[-1].state.data
+        # For legacy state format, the final state message contains the final state of all streams. For per-stream state format,
+        # the complete final state of streams must be assembled by going through all prior state messages received
+        if is_per_stream_state(states_1[-1]):
+            latest_state = construct_latest_state_from_messages(states_1)
+            state_input = list(
+                {"type": "STREAM", "stream": {"stream_descriptor": {"name": stream_name}, "stream_state": stream_state}}
+                for stream_name, stream_state in latest_state.items()
+            )
+        else:
+            latest_state = states_1[-1].state.data
+            state_input = states_1[-1].state.data
+
         for record_value, state_value, stream_name in records_with_state(records_1, latest_state, stream_mapping, cursor_paths):
             assert (
                 record_value <= state_value
             ), f"First incremental sync should produce records younger or equal to cursor value from the state. Stream: {stream_name}"
```

Comment:
Not related to this PR.
I think this message is wrong: 
```suggestion
            ), f"First incremental sync should produce records older or equal to cursor value from the state. Stream: {stream_name}"
```

Reply:
And :
`f"Second incremental sync should produce records younger or equal to cursor value from the state. Stream: {stream_name}"`
Records from first sync: 👴 
Records from second sync: 👶